### PR TITLE
treewide: Qt5.X -> Qt5.15 (when possible)

### PR DESCRIPTION
**Pull request by freezeboy**
###### Motivation for this change

I tried to migrate the software still using Qt5.12 to a newer version, sometimes I had to update the software to make it work.

Among the reasons blocking the packages, I already spotted:

 * Dependency to qtwebkit which is broken with Qt5.15
 * Use of QPainterPath class, apparently the header definition has moved between 5.14 and 5.15
 * Some package build configurations forbid deprecated members use ... and there are such in Qt5.15
 * Other oddities in the code that is not more valid (changing, return types)
 * A few of them compile .. then segfault (digitalbitbox)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by doronbehar — APPROVED** [review]
This is a tremendous work! Did you test each and every package to build and run right?

Overall I approve, but the commit messages sometimes miss some details of other changes, it might help to get this approved but I wouldn't make this a blocker.

**Comment by freezeboy** [comment]
Yes I built and run all the executables, just to check it starts.

For the commit messages, I can see to improve them, at least the ones where an update was needed.

**Comment by doronbehar** [comment]
Well done :clap:.

**Comment by freezeboy** [comment]
Most of the effort is done by my cpu and its fan 😏

**Comment by freezeboy** [comment]
I force pushed with more details in the commit messages, please tell me if you need more

**Comment by doronbehar** [comment]
Pushed after a rebase to fix merge conflicts. Ideally I think every usage of a pinned `libsForQt5..` should be accompanied with a comment. I counted 129 usages of `libsForQt514.callPackage` that come with no comment explaining them.

**Comment by freezeboy** [comment]
Thank you, I will continue to add other apps, and I think we can think to merge this PR, I will start another PR for to continue this process otherwise we will have more and more conflicts

**Comment by freezeboy** [comment]
Ok, I stop for this PR, time for week end :smile:

**Review by jonringer — APPROVED** [review]
diff LGTM
few programs I tried seem to work as intended

cc @ttuegel 
```
https://github.com/NixOS/nixpkgs/pull/102840
1 package marked as broken and skipped:
retroshare

59 packages built:
appcsxcad birdtray calaos_installer chiaki clipgrab cutecom digitalbitbox eagle flameshot golden-cheetah kdiff3 krename kronometer ktorrent libsForQt5.herqq libsForQt5.kde2-decoration libsForQt5.qcsxcad libsForQt5.qtcurve libsForQt512.herqq libsForQt512.kde2-decoration libsForQt512.qcsxcad libsForQt512.qtcurve libsForQt514.herqq libsForQt514.kde2-decoration libsForQt514.qcsxcad libsForQt514.qtcurve minitube openems ostinato pentobi peruse plex-media-player protonmail-bridge psi psi-plus python27Packages.python-csxcad python37Packages.python-csxcad python37Packages.python-openems python38Packages.python-csxcad python38Packages.python-openems qlcplus rsibreak saga samplv1 sdrangel sigil skanlite skrooge solarus solarus-quest-editor sonic-lineup strawberry teamspeak_client tellico virt-manager-qt vnote webcamoid xflr5 zoom-us
```

File: pkgs/data/themes/qtcurve/default.nix
Comment (gnidorah) [inline pkgs/data/themes/qtcurve/default.nix:15]: As an alternative to using unstable version you could add the following patch to release https://github.com/KDE/qtcurve/commit/44e2a35ebb164dcab0bad1a9158b1219a3ff6504

Comment (freezeboy) [reply pkgs/data/themes/qtcurve/default.nix:15]: Well it was the idea, I just have a few other patches that don't look very dangerous.
Do you think it is still important to add just one commit ?

File: pkgs/data/themes/qtcurve/default.nix
Comment (gnidorah) [inline pkgs/data/themes/qtcurve/default.nix:11]:
```suggestion
  version = "2020-06-21";
```
For commit https://github.com/KDE/qtcurve/commit/e534cd04e6acc90c105b492f3fcf99cc6155ba15

File: pkgs/data/themes/qtcurve/default.nix
Comment (gnidorah) [reply pkgs/data/themes/qtcurve/default.nix:15]: Do as you like, just don't forget about version :smiley:

Comment (doronbehar) [reply pkgs/data/themes/qtcurve/default.nix:15]: I too agree with @gnidorah on this - adding a patch is a bit more rigorous and a version change warrants a commit with `<old version> -> <new version>`.

**Comment by jonringer** [comment]
is this still being actively developed?

**Comment by freezeboy** [comment]
If you think the state is ok to be merged, then go, I can work on another PR, I just fixed a few things from various reviewers (and thank all of the btw)

File: pkgs/top-level/all-packages.nix
Comment (jonringer) [inline pkgs/top-level/all-packages.nix:13202]:
```suggestion
  };
```
to fix ofborg

**Comment by doronbehar** [comment]
The last 2 commits should be replaced with a rebase.

**Comment by doronbehar** [comment]
Github sometimes doesn't notify subscribers when a force-push happens (perhaps only for rebases). I think this is ready now, and worth merging.

**Comment by jshholland** [comment]
Should nextcloud-client be included in this? 22167ae45fb27fac1e6aec7d1bf6ce82516fcb12 pinned it to 5.14, but at the moment it segfaults on launch for me. If I switch it to `libsForQt5.callPackage ../applications/networking/nextcloud-client { };` then it launches and runs fine.

**Comment by jnetod** [comment]
https://github.com/NixOS/nixpkgs/pull/96309 fixes qtwebkit on 5.15 too.

**Comment by freezeboy** [comment]
I propose to merge this pr, there are still many other packages to test/fix with qt5.15 we can open a new pr without problem

**Comment by doronbehar** [comment]
I won't mind keep adding packages here, until we get @ttuegel's approval. Perhaps even the qtwebkit could be fixed here, and more qt5.14 pins could be dropped.

**Comment by doronbehar** [comment]
I added the qtwebkit patches for qt515.qtwebkit, and they work with e.g `kmymoney`. @freezeboy unless you wish to not broaden the scope of this PR, I won't mind pushing it here.

**Comment by freezeboy** [comment]
Feel free to complete this PR, I am working on other packages atm

File: pkgs/applications/editors/sigil/default.nix
Comment (ttuegel) [inline pkgs/applications/editors/sigil/default.nix:28]: This is likely to silently break if the package is updated. Let's do something else, like

```suggestion
    sed -i '/^QTLIB_DIR=/ d' src/Resource_Files/bash/sigil-sh_install
```

File: pkgs/top-level/all-packages.nix
Comment (ttuegel) [inline pkgs/top-level/all-packages.nix:15422]: Isn't this an application?

**Review by ttuegel — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix
Comment (freezeboy) [reply pkgs/top-level/all-packages.nix:15422]: Not it is a library, I also had this suprise.
Should I move it to dev/libs ?

**Comment by freezeboy** [comment]
I just force-pushed to apply @ttuegel suggestions

File: pkgs/top-level/all-packages.nix
Comment (doronbehar) [reply pkgs/top-level/all-packages.nix]: It doesn't matter if upstream considers it a library or an application, it matters if we have another qt package that uses it as an input. And we do: `appcsxcad` and `openems`.

**Comment by tobiasBora** [comment]
Any news on this? Would love to see kmymoney back on my system.

**Review by jonringer — APPROVED** [review]
diff LGTM
I don't want to bitrot much longer.
If there's no regressions from master during my nixpkgs-review, I'll merge

**Review by jonringer — COMMENTED** [review]
do you mind rebasing on the lastest master, there's a few conflicts:

```
Auto-merging pkgs/top-level/all-packages.nix
CONFLICT (content): Merge conflict in pkgs/top-level/all-packages.nix
Auto-merging pkgs/top-level/aliases.nix
Auto-merging pkgs/games/pentobi/default.nix
Auto-merging pkgs/development/libraries/qt-5/5.15/default.nix
Removing pkgs/applications/gis/saga/opencv4.patch
Removing pkgs/applications/gis/saga/finite.patch
Removing pkgs/applications/gis/saga/clang_patch.patch
Automatic merge failed; fix conflicts and then commit the result.
https://github.com/NixOS/nixpkgs/pull/102840 failed to build
```

**Review by esclear — APPROVED** [review]

**Comment by freezeboy** [comment]
Rebase done

**Comment by jonringer** [comment]
LGTM

failures exist on master
```
https://github.com/NixOS/nixpkgs/pull/102840
2 packages marked as broken and skipped:
retroshare retroshare06

3 packages failed to build:
kmymoney obs-ndi xflr5

70 packages built:
amarok appcsxcad birdtray calaos_installer chiaki clipgrab communi csound-qt cutecom digitalbitbox eagle flameshot fritzing golden-cheetah herqq kdiff3 kgraphviewer krename kronometer ktorrent libsForQt5.herqq libsForQt5.kde2-decoration qcsxcad qtcurve libsForQt512.herqq libsForQt512.kde2-decoration libsForQt512.qcsxcad libsForQt512.qtcurve libsForQt514.kde2-decoration libsForQt514.qcsxcad libsForQt514.qtcurve luminanceHDR massif-visualizer minitube obs-move-transition obs-studio obs-v4l2sink obs-wlrobs openems ostinato pentobi peruse plex-media-player protonmail-bridge psi psi-plus python27Packages.python-csxcad python37Packages.python-csxcad python37Packages.python-openems python38Packages.python-csxcad python38Packages.python-openems qlcplus rsibreak rssguard saga samplv1 sdrangel sigil skanlite skrooge solarus solarus-quest-editor sonic-lineup strawberry teamspeak_client tellico virtmanager-qt vnote webcamoid zoom-us
```

**Comment by freezeboy** [comment]
I guess `obs-ndi` is normal as the rest of the stack now use qt5.15, I can change its Qt version too, but I won't be able to build it anyway

I start looking the other errors

**Comment by jonringer** [comment]
> I guess `obs-ndi` is normal as the rest of the stack now use qt5.15, I can change its Qt version too, but I won't be able to build it anyway
> 
> I start looking the other errors

either way, this is still an improvement. Thanks! :)

**Comment by doronbehar** [comment]
> 3 packages failed to build:
> kmymoney obs-ndi xflr5

That's a regression from the last time I tried to build kmymoney - it fails due to xmlsec not building:

https://gist.github.com/doronbehar/4b34941feb73855eb307b3470d12dff2

**Comment by doronbehar** [comment]
@freezeboy when you rebased it you forced pushed and forgot some commits of mine :(. For instance qtwebkit was fixed there.

**Comment by doronbehar** [comment]
Follow up PR: https://github.com/NixOS/nixpkgs/pull/104474

kmymoney needs https://github.com/NixOS/nixpkgs/pull/104472 in order to work.

**Pull request merged**
